### PR TITLE
Testing blog post

### PR DESCRIPTION
**Pull request by otoolep**

### CODE REVIEW

**Comment by otoolep** [comment]
cc @mjdesa

File: content/blog/2015/10/20/testing_tsm.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+![](/img/blog/testing_tsm/100b-1hrt.png)
+
+Some interesting features are present. Write load is steady, as is CPU load. Disk usage reached a steady-state as the retention enforcement deleted data at about the rate incoming data was being indexed. The sawtooth pattern in disk usage is a mixture of compaction performed by the tsm1 engine, as well as retention enforcement. Interestingly InfluxDB RSS correlates closely with disk usage, which makes sense to us. The software memory-maps data on disk, so as data on disk is deleted, memory usage declines. But memory usage is generably steady over the long run, at about 30% total physical RAM on the host machine. This tells us this software does not suffer any detectable memory leaks. (The short spike at the very left of each graph is an aborted run.)
```
Comment (toddboom) [inline content/blog/2015/10/20/testing_tsm.md]: generably => generally

**Comment by toddboom** [comment]
@otoolep @mjdesa this is a fantastic write-up. one small nit, then merge whenever you're ready!

**Comment by otoolep** [comment]
Fixed, I'll let you merge as you will be closer to the desired publication schedule.

**Comment by toddboom** [comment]
@otoolep awesome, thanks - i went ahead and merged because it won't get generated into the index until the `publishdate` arrives.

**Comment by otoolep** [comment]
Nice, I was wondering why I had to artificially push the publish date back to get it working with my hugo server.

**Comment by toddboom** [comment]
Only for a little while longer...

File: content/blog/2015/10/20/testing_tsm.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+While an InfluxDB system is under load we record various metrics about the host machine during the entire test run. Most importantly we record disk IO, memory usage, CPU load, and the [resident set size](https://en.wikipedia.org/wiki/Resident_set_size) of the InfluxDB process. During and after a test the results are shared with the team, and with the help of our Grafana dashboads, we study the results for problems, and (hopefully!) confirm our software is working as expected.
+
+## Testing the TSM1 Engine
+
+We recently starting testing the [new tsm1 storage engine](https://influxdb.com/blog/2015/10/07/the_new_influxdb_storage_engine_a_time_structured_merge_tree.html). A recent test ran for about 8 hours and involved writing 100 billion points to a single InfluxDB node, across 1000s of different series. The target retention policy also had a duration of 1 hour, so we could test those code paths too -- since old data would be deleted hourly as new data was being indexed. The Grafana dashboard, with results of the complete test run, is shown below.
```
Comment (beckettsean) [inline content/blog/2015/10/20/testing_tsm.md]: This implies that we wrote 100 billion points to the database in 8 hours, which would be ~3.5 million points per second. That cannot be true (yet).

100,000,000,000 / (3600*8) = 3,472,222.2...

**Comment by beckettsean** [comment]
@otoolep @mjdesa before we publish please clarify the rate for the test

**Comment by mjdesa** [comment]
@beckettsean good catch

File: content/blog/2015/10/20/testing_tsm.md (hunk 2 above)
Comment (otoolep) [inline content/blog/2015/10/20/testing_tsm.md]: Actually, yeah, that must be incorrect. I can't recall exactly why the test only ran for that long, so let me remove the details.

**Comment by otoolep** [comment]
I removed the detail, since this test is just about resource usage, not about specific numbers of points. Do we have to push again to publish?

**Pull request merged**